docs/standard/security/cryptographic-signatures.md

```diff
@@ -30,129 +30,83 @@ This topic explains how to generate and verify digital signatures using classes
+            rsaFormatter.SetHashAlgorithm(NameOf(SHA256))
 
-        'Set the hash algorithm to SHA1.
-        rsaFormatter.SetHashAlgorithm("SHA1")
+            signedHash = rsaFormatter.CreateSignature(hash)
+        End Using
 
-        'Create a signature for hashValue and assign it to
-        'signedHashValue.
-        signedHashValue = rsaFormatter.CreateSignature(hashValue)
+        ' The sharedParameters, hash, and signedHash are used to later verify the signature.
     End Sub
 End Module
+
 ```
 
 ```csharp
-using System;
 using System.Security.Cryptography;
+using System.Text;
 
-class Class1
-{
-   static void Main()
-   {
-      //The hash value to sign.
-      byte[] hashValue = {59,4,248,102,77,97,142,201,210,12,224,93,25,41,100,197,213,134,130,135};
+using SHA256 alg = SHA256.Create();
 
-      //The value to hold the signed value.
-      byte[] signedHashValue;
+byte[] data = Encoding.ASCII.GetBytes("Hello, from the .NET Docs!");
+byte[] hash = alg.ComputeHash(data);
 
-      //Generate a public/private key pair.
-      RSA rsa = RSA.Create();
+RSAParameters sharedParameters;
+byte[] signedHash;
 
-      //Create an RSAPKCS1SignatureFormatter object and pass it the
-      //RSA instance to transfer the private key.
-      RSAPKCS1SignatureFormatter rsaFormatter = new RSAPKCS1SignatureFormatter(rsa);
+// Generate signature
+using (RSA rsa = RSA.Create())
+{
+    sharedParameters = rsa.ExportParameters(false);
 
-      //Set the hash algorithm to SHA1.
-      rsaFormatter.SetHashAlgorithm("SHA1");
+    RSAPKCS1SignatureFormatter rsaFormatter = new(rsa);
+    rsaFormatter.SetHashAlgorithm(nameof(SHA256));
 
-      //Create a signature for hashValue and assign it to
-      //signedHashValue.
-      signedHashValue = rsaFormatter.CreateSignature(hashValue);
-   }
+    signedHash = rsaFormatter.CreateSignature(hash);
 }
+
+// The sharedParameters, hash, and signedHash are used to later verify the signature.
 ```
 
 ## Verifying Signatures
 
 To verify that data was signed by a particular party, you must have the following information:
 
 - The public key of the party that signed the data.
-
 - The digital signature.
-
 - The data that was signed.
-
 - The hash algorithm used by the signer.
 
 To verify a signature signed by the <xref:System.Security.Cryptography.RSAPKCS1SignatureFormatter> class, use the <xref:System.Security.Cryptography.RSAPKCS1SignatureDeformatter> class. The <xref:System.Security.Cryptography.RSAPKCS1SignatureDeformatter> class must be supplied the public key of the signer. For RSA, you will need the values of the modulus and the exponent to specify the public key. (The party that generated the public/private key pair should provide these values.) First create an <xref:System.Security.Cryptography.RSA> object to hold the public key that will verify the signature, and then initialize an <xref:System.Security.Cryptography.RSAParameters> structure to the modulus and exponent values that specify the public key.
```

Comment:
Why are we removing the modulus/exponent example? They're still mentioned on line 100.

Reply:
Those are the only two fields that are required on Windows, the others are optional. Instead, I'm trying to highlight that the creation of the signature should instead export its parameters, and those parameters should be imported later by the verifying RSA. The original example didn't show where those values came from, or how to correctly provide them. Does that make sense?